### ISSUE TITLE
Average Duration can (under certain conditions) be wrong

### ISSUE THREAD

**Issue by kelson42** [openzim/zimfarm#1031]
Looking at https://farm.openzim.org/recipes/cest-pas-sorcier_fr_astronomie, it's written `Duration: 40 minutes, 40 seconds`
![Image](https://github.com/user-attachments/assets/9d2f8dca-deb7-4859-a9fc-b09db9b1caf1)

But this recipe had only two (failed) attempts/tasks and each of them has taken only a few minutes.

Therefore the ~ 40 minutes duration can not be valid.

Considering that AFAIK this value should be the longest duration for the last tracked successful attempts, I guess that no value should be computed/displayed at all for the "Duration" in such a scenario.

**Comment by rgaudin** [comment]
I believe this duration is coming from the schedule that has been cloned into this one. It's actually indicated by the `default` and the date. Scheduler  kinda requires an existing duration which is why we have a default.

**Comment by benoit74** [comment]
Yep, I'm pretty sure it comes from the clone. I'm quite sure it is solved once we have a succesful run. Anyway, this should still be fixed.

**Comment by elfkuzco** [comment]
@benoit74 , could you take a look at this at it should not happen anymore? In #1360, the scraper duration now computes between when a worker started and when it succeeded, thus making this unlikely.

**Comment by benoit74** [comment]
I confirm values in DB :

| name | default | value | name |
| :--- | :--- | :--- | :--- |
| cest-pas-sorcier\_fr\_astronomie | true | 2678400 | null |
| cest-pas-sorcier\_fr\_astronomie | false | 117 | pixelmemory |
| cest-pas-sorcier\_fr\_astronomie | false | 1053 | badger1 |
| cest-pas-sorcier\_fr\_astronomie | false | 255 | athena18 |

And values in UI :

<img width="1512" height="544" alt="Image" src="https://github.com/user-attachments/assets/f9fd24fa-01fb-4b90-b706-f147e79857d7" />

Are aligned.

It is a bit hard to tell if the "2 minutes" on pixel memory worker is correct or not, we do not have a "2 minutes" succeeded run in the history. 

In order to close the task, @elfkuzco can you please double check in codebase when durations are created / updated in DB (it should be only when it task has succeeded)?

**Comment by elfkuzco** [comment]
Um, "Scraper Duration" is how long it took to go from `started` to `scraper_completed`. If you click on the task that succeeded with `4 minutes` you would see it takes 2 minutes to get to `scraper_completed` and an additional `2 minutes` to get to `succeeded`. Durations are updated only on `scraper_completed`
https://github.com/openzim/zimfarm/blob/ef91be84720e4a60936a7482261859f1717b6803/backend/src/zimfarm_backend/common/utils.py#L182-L185

**Comment by benoit74** [comment]
Indeed, I read the wrong line. Sorry for that. 

Looks like we have no explanation for why it might have been wrong at some point, but now it is correct.

Closing then.
